### PR TITLE
[master < T0663-MG] Add kafka documentation

### PR DESCRIPTION
**Pull request by kostasrim**
This is in WIP. Please do not review yet.

### CODE REVIEW

File: docs/transformation-modules/api/c-api.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+## Classes
+
+|                | Name           |
+| -------------- | -------------- |
+| struct | **mgp_label** <br/>Label of a vertex.  |
+| struct | **mgp_edge_type** <br/>Type of an edge.  |
+| struct | **mgp_property** <br/>Reference to a named property value.  |
+| struct | **mgp_vertex_id** <br/>ID of a vertex; valid during a single query execution.  |
+| struct | **mgp_edge_id** <br/>ID of an edge; valid during a single query execution.  |
+
```
Comment (antonio2368) [inline docs/transformation-modules/api/c-api.md:22]: Why is this here?

File: docs/transformation-modules/api/c-api.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+## Types
+
+|                | Name           |
+| -------------- | -------------- |
+| typedef void (*mgp_trans_cb)(const struct mgp_messages *, const struct mgp_graph *, struct mgp_result *, struct mgp_memory *); |
+  **[mgp_trans_cb](#typedef-mgp_trans_cb)** <br/>Entry-point for a transformation with a fixed result type |
```
Comment (antonio2368) [inline docs/transformation-modules/api/c-api.md:28]: Escape all the asterix as you said.
Also, these 2 lines need to be in a single line so this is in a single row in a table. Right now it's rendered into 2 rows.

File: docs/transformation-modules/api/c-api.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+## Functions
+
+|                | Name           |
+| -------------- | -------------- |
+| size_t | **[mgp_message](#function-mgp_messages_size)**(const struct mgp_messages \*messages) <br/>Get the number of messages contained in the messages list. |
```
Comment (antonio2368) [inline docs/transformation-modules/api/c-api.md:34]:
```suggestion
| size_t | **[mgp_messages_size](#function-mgp_messages_size)**(const struct mgp_messages \*messages) <br/>Get the number of messages contained in the messages list. |
```

File: docs/transformation-modules/api/c-api.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+| size_t | **[mgp_message_payload_size](#function-mgp_message_payload)**(const struct mgp_message \*message) <br/> Get the payload size of message. |
+| const char * | **[mgp_message_payload](#function-mgp_message_payload)**(const struct mgp_message \*message) <br/> Get the payload of messages as a byte array.|
+| const char * | **[mgp_message_topic_name](#function-mgp_message_topic_name)**(const struct mgp_message \*message) <br/> Get the topic name of message. |
+| size_t | **[mgp_message_key_size](#function-mgp_message_key_size)**(const struct mgp_message \*message) <br/> Get key size of message. |
+| const char * | **[mgp_message_key](#function-mgp_message_key)**(const struct mgp_message \*message) <br/> Get key of message as a byte array. |
+| int64_t | **[mgp_message](#function-mgp_message_timestamp)**(const struct mgp_message \*message) <br/> Get the timestamp of message. |
+| const struct mgp_message * | **[mgp_messages_at](#function-mgp_messages_at)**(const struct mgp_messages \*messages, size_t idx) <br/> Get the mgp_message at index idx. |
```
Comment (antonio2368) [inline docs/transformation-modules/api/c-api.md:41]: Move this below mgp_messages_size

File: docs/transformation-modules/api/python-api.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+## `mgp.transformation(func)`
+Register func as a transformation of the current module.
+
+transformation is meant to be used as a decorator function to register a transformation 
+within the current module. The registered func needs to be a callable which either accepts
+a single argument of type Messages or two arguments of type TransCtx and Messages respectively. 
+The return type is fixed and it should be of the form:
+Record(query=str, parameters=mgp.Map). Multiple records can be
+produced by returning an iterable of them. Registering generator functions
+is currently not supported.
```
Comment (antonio2368) [inline docs/transformation-modules/api/python-api.md:24]:
```suggestion
`mgp.transformation` is meant to be used as a decorator function to register a transformation 
within the current module. The registered `func` needs to be a callable which either accepts
a single argument of type `Messages` or two arguments of type `TransCtx` and `Messages` respectively. 
The return type is fixed and it should be of the form:
`Record(query=str, parameters=mgp.Map)`. Multiple records can be
produced by returning an iterable of them. Registering generator functions
is currently not supported.
```

And do the same every time with the backtick when you see something which is code specific in the middle of the text.

File: docs/transformation-modules/api/python-api.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+Any errors can be reported by raising an Exception.
+
+## `class mgp.Message(messages)`
+Bases: `object`
+
+Represents a single message. 
+You should not globally store Message.
+
+### `is_valid()`
+Returns true if self is valid and can be accessed.
```
Comment (antonio2368) [inline docs/transformation-modules/api/python-api.md:60]:
```suggestion
Returns true if message is valid and can be accessed.
```

File: docs/transformation-modules/api/python-api.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+Raises an InvalidMessageError if is_valid() is false.
+
+## `class mgp.Messages(messages)`
+Bases: `object`
+
+Represents a list of messages passed to a transformation. 
+You should not globally store messages.
+
+### `is_valid()`
+Returns true if self is valid and can be accessed.
```
Comment (antonio2368) [inline docs/transformation-modules/api/python-api.md:85]:
```suggestion
Returns true if messages object is valid and can be accessed.
```

File: docs/transformation-modules/api/python-api.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+### `total_messages()`
+Returns the number of messages contained.
+Raises InvalidMessagesError if is_valid() is false.
+
+### `message_at(id)`
+Returns the underline message at index id.
```
Comment (antonio2368) [inline docs/transformation-modules/api/python-api.md:92]:
```suggestion
Returns the underlying message at index `id`.
```

File: docs/transformation-modules/introduction.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
+---
```
Comment (antonio2368) [inline docs/transformation-modules/introduction.md:1]: you can combine these two files (`introduction.md` and `overview.md` into one called `overview.md`)

File: docs/reference-guide/streams/streams.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+## Creating a stream (query)
+The general syntax for creating a stream is:
+```cypher
+CREATE STREAM <stream name>
+  WITH TOPICS (['<topic1>', '<topic2>'])
+  WITH TRANSFORM <transform procedure>
+  [WITH CONSUMER_GROUP <consumer group name>]
+  [BATCH_INTERVAL <milliseconds>]
+  [BATCH_SIZE <size>];
```
Comment (antonio2368) [inline docs/reference-guide/streams/streams.md:26]: remove `WITH` everywhere

File: docs/reference-guide/streams/streams.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+- status
+- consumer group id
+- batch interval
+- batch size
+
+## Test connection (query)
+```cypher
+TEST STREAM <stream name> [LIMIT <count> BATCHES];
+```
+Tests the connection of `<stream name>` with `<count>` number of batches.
```
Comment (antonio2368) [inline docs/reference-guide/streams/streams.md:90]: Replace the word `test` with `check`. Same thing in the query and don't forget to add the new timeout config.

File: sidebarsMemgraph.js [hunk 12]
```diff
@@ -127,6 +127,17 @@ module.exports = {
...
+          type: 'category',
+          label: 'Streams',
+          items: ['reference-guide/streams/transformation-modules/overview',
+            {
+              type: 'category',
+              label: 'API',
+              items: ['reference-guide/streams/transformation-modules/api/c-api',
+                'reference-guide/streams/transformation-modules/api/python-api',],
+            }],
+        },
```
Comment (antonio2368) [inline sidebarsMemgraph.js:140]:
```suggestion
        {
          type: 'category',
          label: 'Streams',
          items: ['reference-guide/streams/streams',
          {
            type: 'category',
            label: 'Transformation Modules',
            items: ['reference-guide/streams/transformation-modules/overview',
              {   
                type: 'category',
                label: 'API',
                items: ['reference-guide/streams/transformation-modules/api/c-api',
                  'reference-guide/streams/transformation-modules/api/python-api',],
              }], 
          }], 
        },
```

**Review by antonio2368 — COMMENTED** [review]
Looks really, really nice!

File: docs/reference-guide/streams/streams.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+must 
+- Manage a stream via a query
+- Provide a user defined transformation module
+
+More information about transformation modules can be found
+**[here](./transformation-modules/api/c-api.md)**
+
+The rest of this section describes how to manage streams with memgraph.
+
+## Creating a stream (query)
```
Comment (antonio2368) [inline docs/reference-guide/streams/streams.md:18]: I would remove `(query)` everywhere, it just makes things harder to read.

File: docs/reference-guide/streams/streams.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+## Check stream (query)
+```cypher
+CHECK STREAM <stream name> [LIMIT <count> BATCHES] [TIMEOUT <milliseconds>] ;
+```
+Checks the stream with name `<stream name>` with  `<count>` number of batches.
+
+After `<count>` batches are processed the transformation result is returned.
+The result can be empty if the batch interval is reached.
+`TIMEOUT` is measured in milliseconds.
```
Comment (antonio2368) [inline docs/reference-guide/streams/streams.md:94]: document the default value

Comment (kostasrim) [reply docs/reference-guide/streams/streams.md:94]: Just did. Same for `<count>` which defaults to 1.

**Review by g-despot — APPROVED** [review]
